lib/Doctrine/ORM/EntityManager.php

```diff
@@ -826,28 +826,9 @@ public function initializeObject($obj)
      */
     public static function create($conn, Configuration $config, EventManager $eventManager = null)
     {
-        if ( ! $config->getMetadataDriverImpl()) {
-            throw ORMException::missingMappingDriverImpl();
-        }
-
-        switch (true) {
-            case (is_array($conn)):
-                $conn = \Doctrine\DBAL\DriverManager::getConnection(
-                    $conn, $config, ($eventManager ?: new EventManager())
-                );
-                break;
-
-            case ($conn instanceof Connection):
-                if ($eventManager !== null && $conn->getEventManager() !== $eventManager) {
-                     throw ORMException::mismatchedEventManager();
-                }
-                break;
-
-            default:
-                throw new \InvalidArgumentException("Invalid argument: " . $conn);
-        }
+        $emf = new EntityManagerFactory;
```

Comment:
Since the code is so thightly coupled together with the entity manager, I don't see a real advantage in having a separate factory class instead of just hardcoded static method logic here.